src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -462,19 +474,31 @@ module Ledger = struct
             ~ledger_name_prefix config
         in
         match tar_path with
-        | Some tar_path -> (
+        | Some tar_file -> (
+            let extracted_folder =
+              Tar.filename_without_extension @@ Filename.basename tar_file
+            in
+            let extracted_path = genesis_dir ^/ extracted_folder in
+            let load_ledger_spec =
+              match padded_accounts_opt with
+              | Some accounts ->
+                  Accounts { accounts; link_path = Some extracted_path }
+              | None ->
+                  Tar { tar_file; extracted_path }
+            in
```

Comment:
Does this work? The old `load_ledger_by_spec` looks like always clears and recreates `dirname`, then extracts the tar into `dirname`. Here, I think the tar file is never extracted for an `Accounts {}` spec.

Reply:
You might fix this later on.

Reply:
Or I misunderstood the code, which is very possible

Reply:
Wait, is this okay because `Genesis_ledger.Make` ignores the supplied `accounts` when given a `` `Path `` parameter? Very tricky.

Reply:
Yeah, looks like I missed that as well, thanks :)